New float16 failures in 32-bit Windows wheel build job for cp314t

There are new failures in a full set of wheel builds on the latest `main` branch for `float16`, specifically for 32-bit Windows with a on cp314t interpreter. 

A lot of the failures seem to have to do with `sort`/`partition`, see [this CI log](https://github.com/numpy/numpy-release/actions/runs/17965024295/job/51113515921).

```
2025-09-24T08:46:03.2017662Z ..\venv-test\Lib\site-packages\numpy\_core\tests\test_multiarray.py:10653: 
2025-09-24T08:46:03.2018258Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2025-09-24T08:46:03.2018584Z 
2025-09-24T08:46:03.2018731Z kth = np.float16(-0.3762), k = np.int32(302)
2025-09-24T08:46:03.2019255Z arr_part = array([-0.2607 , -0.2607 , -0.2607 ,  0.1748 , -0.4316 , -0.4316 ,
2025-09-24T08:46:03.2019779Z        -0.3027 ,  0.1748 , -0.309  , -0.309  , -0.3... 0.4526 ,
2025-09-24T08:46:03.2020387Z         0.4526 ,  0.4526 , -0.2451 , -0.2451 ,  0.4526 ,  0.4526 ,
2025-09-24T08:46:03.2020815Z         0.4526 ,  0.4526 ], dtype=float16)
2025-09-24T08:46:03.2021063Z 
2025-09-24T08:46:03.2021219Z     def assert_arr_partitioned(kth, k, arr_part):
2025-09-24T08:46:03.2021653Z >       assert_equal(arr_part[k], kth)
2025-09-24T08:46:03.2022034Z E       AssertionError: 
2025-09-24T08:46:03.2022344Z E       Items are not equal:
2025-09-24T08:46:03.2022696Z E        ACTUAL: np.float16(0.1777)
2025-09-24T08:46:03.2023072Z E        DESIRED: np.float16(-0.3762)
2025-09-24T08:46:03.2023354Z 
2025-09-24T08:46:03.2023618Z arr_part   = array([-0.2607 , -0.2607 , -0.2607 ,  0.1748 , -0.4316 , -0.4316 ,
2025-09-24T08:46:03.2024184Z        -0.3027 ,  0.1748 , -0.309  , -0.309  , -0.3... 0.4526 ,
2025-09-24T08:46:03.2024654Z         0.4526 ,  0.4526 , -0.2451 , -0.2451 ,  0.4526 ,  0.4526 ,
2025-09-24T08:46:03.2025096Z         0.4526 ,  0.4526 ], dtype=float16)
2025-09-24T08:46:03.2025483Z k          = np.int32(302)
2025-09-24T08:46:03.2025806Z kth        = np.float16(-0.3762)

... <warnings>

025-09-24T08:46:03.2307720Z inf location mismatch:
2025-09-24T08:46:03.2307825Z  ACTUAL: array([-0.3323,     inf], dtype=float16)
2025-09-24T08:46:03.2307939Z  DESIRED: array(-0.3323, dtype=float16)
2025-09-24T08:46:03.2308141Z FAILED _core/tests/test_multiarray.py::test_partition_fp[float16-N3] - AssertionError: 
2025-09-24T08:46:03.2308146Z 
2025-09-24T08:46:03.2308161Z 
2025-09-24T08:46:03.2308224Z inf location mismatch:
2025-09-24T08:46:03.2308356Z  ACTUAL: array([-0.3323, -0.3323, -0.3323,     inf], dtype=float16)
2025-09-24T08:46:03.2308438Z  DESIRED: array(-0.3323, dtype=float16)
2025-09-24T08:46:03.2309086Z FAILED _core/tests/test_multiarray.py::test_partition_fp[float16-N4] - AssertionError: 
2025-09-24T08:46:03.2309183Z Items are not equal:
2025-09-24T08:46:03.2309300Z  ACTUAL: np.float16(0.4185)
2025-09-24T08:46:03.2309401Z  DESIRED: np.float16(inf)
2025-09-24T08:46:03.2309718Z FAILED _core/tests/test_multiarray.py::test_partition_fp[float16-N5] - AssertionError: 
2025-09-24T08:46:03.2309809Z Items are not equal:
2025-09-24T08:46:03.2309917Z  ACTUAL: np.float16(-0.0886)
```

Does this ring a bell for anyone? Recent PRs that touched `float16` and/or `sort` in a nontrivial way?

I’m seeing a lot of warnings (I’m working on suppressing them), but one of them caught my attention and might be related to this runtime error seems after #29129:

```Bash
lowlevel_strided_loops.c.src(767): warning C4244: 'return': conversion from 'double' to 'npy_uint64', possible loss of data ... caused by #29129
```

https://github.com/numpy/numpy/blob/e7276da07f6fc69cc2947f4dc67a87472e4b3a8b/numpy/_core/src/multiarray/lowlevel_strided_loops.c.src#L767

What happens here is ToDoubleBits() → uint64_t → double → uint64_t (cast in the return). That intermediate double cast shouldn't be here and should be removed; GCC/Clang likely optimized it out. I’ll finish clearing the warnings and running the tests and dive more if it needed.

--- a/numpy/_core/meson.build
+++ b/numpy/_core/meson.build
@@ -867,7 +867,9 @@ foreach gen_mtargets : [
   [
     'x86_simd_qsort_16bit.dispatch.h',
     'src/npysort/x86_simd_qsort_16bit.dispatch.cpp',
-    use_intel_sort ? [AVX512_SPR, AVX512_ICL] : []
+    # Do not enable AVX-512 on MSVC 32-bit (x86): it’s buggy there;
+    # Ref: NumPy issue numpy/numpy#29808
+    use_intel_sort and not (cc.get_id() == 'msvc' and cpu_family == 'x86') ? [AVX512_SPR, AVX512_ICL] : []
   ],
   [
     'highway_qsort.dispatch.h',

